interview-questions.md

```diff
@@ -9,41 +9,41 @@ Then, research the question to expand on your answer. Even if you feel you have
 
 1. What is `this` in JavaScript?
 
-  Your answer:
+  Your answer: .this is how you call an object.
 
-  Researched answer:
+  Researched answer: a key word that calls an object invoking the object into the code or calling on to perform an execution.
 
 
 
 2. What is React? Why would you use it?
 
-  Your answer:
+  Your answer: coding langauge made by facebook that creates sections of html that refresh and update data on the section its used on to keep debugging seemless
 
-  Researched answer:
+  Researched answer: A scripting frame work that allows you to access a code library. Made by Facebook that allows you to use their created UI's. Without storing it as a whole on your local nextwork or make it yourself.
 
 
 
 3. Which lifecycle method is required in a React class component?
 
-  Your answer:
+  Your answer: render() is required because it outputs the input to the DOM.
 
-  Researched answer:
+  Researched answer: render() is essential, it takes two arguments and is required in class components.
 
 
 
 4. What is JSX? What is one notable difference between HTML and JSX?
 
-  Your answer:
+  Your answer: A syntax extension to javascript. I would say the notable diffrence its that if JSX was html it would be called 
 
-  Researched answer:
+  Researched answer: A syntax extension to javascript. It stands for javascript XML. It allows you to write HTML, in react in javascript and place them in the DOM then converts HTML tags into react elements.
 
 
 
 5. What is yarn? What file(s) are modified in a React application when you run yarn?
 
-  Your answer:
+  Your answer: Yarn is a unpacking frame work, that allows you to work with react. In a form your computer can handle allowing you to access the code library, without storing it as a whole on your local nextwork. When you open a yarn lock file before running the proper commands. If you unpack it you will get a yarnlock file and a readme.md file if cloned properly. Then after running yarn the html and js files unlock leaving you to only unlock the package.json.
 
-  Researched answer:
+  Researched answer: An alternative to npm made by facebook. A package manager for node.js. that has speed and consistency than also doubles as a project manager like git hub.
```

Comment:
Great work in this section!

Reply:
thank you i'm making sure to read multiple articles on the topics now and study them before hand